service/src/main/java/tds/session/repositories/SessionRepository.java

```diff
@@ -18,6 +19,14 @@
     Optional<Session> findSessionById(final UUID id);
```

Comment:
I had a bit trouble following.  I think we should remove this method all together, use the varargs one in the service and do the size check there.